src/Mod/Fem/App/FemMesh.cpp

```diff
@@ -1172,6 +1172,433 @@ void FemMesh::readNastran(const std::string &Filename)
+            hexa_element.push_back(atoi(line1.substr(40,48).c_str()));
+            hexa_element.push_back(atoi(line1.substr(48,56).c_str()));
+            hexa_element.push_back(atoi(line1.substr(56,64).c_str()));
+            hexa_element.push_back(atoi(line1.substr(64,72).c_str()));
+            
+            hexa_element.push_back(atoi(line2.substr(8,16).c_str()));
+            hexa_element.push_back(atoi(line2.substr(16,24).c_str()));
+
+            all_elements.push_back(hexa_element);
+        }
+        else if (!nastran_free_format && line1.find("CHEXA2")!= std::string::npos)
+        {
+            //d011121a.inp    
+            //CHEXA1  1       200     1       2       13      12      101     102     +SOL1   
+            //+SOL1   113     112                                                             
+            tetra_element.clear();
+            std::getline(inputfile,line2);
+            unsigned int id = atoi(line1.substr(8,16).c_str());
+            element_type.push_back(382);
+            int offset = 0;
+
+            element_id.push_back(id);
+            hexa_element.push_back(atoi(line1.substr(24,32).c_str()));
+            hexa_element.push_back(atoi(line1.substr(32,40).c_str()));
+            hexa_element.push_back(atoi(line1.substr(40,48).c_str()));
+            hexa_element.push_back(atoi(line1.substr(48,56).c_str()));
+            hexa_element.push_back(atoi(line1.substr(56,64).c_str()));
+            hexa_element.push_back(atoi(line1.substr(64,72).c_str()));
+            
+            hexa_element.push_back(atoi(line2.substr(8,16).c_str()));
+            hexa_element.push_back(atoi(line2.substr(16,24).c_str()));
+
+            all_elements.push_back(hexa_element);
+        }
+
+// free format
+        else if (nastran_free_format && line1.find("GRID")!= std::string::npos ) //We found a Grid line
+        {
+            //Base::Console().Log("Found a free format GRID\n");
+            char_separator<char> sep(",");
+            tokenizer<char_separator<char> > tokens(line1, sep);
+            token_results.assign(tokens.begin(),tokens.end());
+            if (token_results.size() < 3)
+                continue;//Line does not include Nodal coordinates
+            nodal_id.push_back(atoi(token_results[1].c_str()));
+            current_node.x = atof(token_results[3].c_str());
+            current_node.y = atof(token_results[4].c_str());
+            current_node.z = atof(token_results[5].c_str());
+            vertices.push_back(current_node);
+        }
+        else if (nastran_free_format && line1.find("CTETRA")!= std::string::npos)
+        {
+            //Base::Console().Log("Found a CTETRA\n");
+            tetra_element.clear();
+            //Lets extract the elements
+            //As each Element Line consists of two subsequent lines as well
+            //we have to take care of that
+            //At a first step we only extract Quadratic Tetrahedral Elements
```

Comment:
`s/At/As/`

Reply:
Some elements only have one line input. A few have two or more lines.
More test cases are about 2d. Very little 3d cases.

Reply:
You just have a typo on line 1437: substitute the word "As" for the word "At"

Reply:
Thanks. Didn't get it first. 

I thought he was asking something, but it was the code I copied and didn't checked.

https://github.com/FreeCAD/FreeCAD/blob/master/src/Mod/Fem/App/FemMesh.cpp#L1052

Reply:
That's a shame. It looks like you removed the helpful comment altogether...? Comments in code are helpful for other devs, just sayin'

Reply:
That's the part I based on. It only implemented one type at that time which is not true  for my case. I have one comment line at its beginning, consistence with my other types. Since I have no demo case for this type, have to wait when run across one to add the sample data lines for it as I did for other types.